typespec-extension/src/external-schemas.ts

```diff
@@ -108,53 +115,161 @@ export function createPollOperationDetailsSchema(schemas: Schemas, stringSchema:
-  let fileDetailsSchema = fileDetailsMap.get(schemaName);
-  if (!fileDetailsSchema) {
-    fileDetailsSchema = new ObjectSchema(schemaName, 'The file details for the "' + filePropertyName + '" field.', {
-      language: {
-        default: {
-          namespace: namespace,
-        },
-        java: {
-          namespace: getJavaNamespace(namespace),
-        },
-      },
-      serializationFormats: [KnownMediaType.Multipart],
-    });
-    fileDetailsSchema.serializationFormats;
-    schemas.add(fileDetailsSchema);
-    fileDetailsSchema.addProperty(
-      new Property("content", "The content of the file.", binarySchema, {
-        required: true,
-        nullable: false,
-        readOnly: false,
-      }),
-    );
-    fileDetailsSchema.addProperty(
-      new Property("filename", "The filename of the file.", stringSchema, {
-        required: false,
-        nullable: false,
-        readOnly: false,
-      }),
-    );
-    fileDetailsSchema.addProperty(
-      new Property("contentType", "The content-type of the file.", stringSchema, {
-        required: false,
-        nullable: false,
-        readOnly: false,
-        clientDefaultValue: "application/octet-stream",
-      }),
-    );
-    fileDetailsMap.set(schemaName, fileDetailsSchema);
+  if (property.type.kind === "model") {
+    // property.type is File, use name and properties from property.type for the File schema
+    /*
+    Current logic:
+    - Class name suffix "FileDetails"
+    - No class hierarchy for File
+    - File has 3 properties: "content", "filename", "contentType" (Note that it is "contents" in TypeSpec)
+    - No adjustment on "content" property, it is always BinaryData and required
+    - Allow constant type for "filename" and "contentType" (to be discussed for other types e.g. enum)
+    - Allow required for "filename" and "contentType"
+     */
+    const filePropertyName = property.name;
+    const fileSchemaName = property.type.name;
+    const schemaName = getFileSchemaName(fileSchemaName);
+    let fileDetailsSchema = fileDetailsMap.get(schemaName);
+    if (!fileDetailsSchema) {
+      const typeNamespace = getNamespace(property.type.__raw) ?? namespace;
+      fileDetailsSchema = createFileDetailsSchema(schemaName, filePropertyName, typeNamespace, schemas);
+
+      // description if available
+      if (property.type.description) {
+        fileDetailsSchema.summary = property.type.description;
+      }
+      if (property.type.details) {
+        fileDetailsSchema.language.default.description = property.type.details;
+      }
+      // crossLanguageDefinitionId
+      (fileDetailsSchema as CrossLanguageDefinition).crossLanguageDefinitionId =
+        property.type.crossLanguageDefinitionId;
+
+      let contentTypeProperty;
+      let filenameProperty;
+
+      // find "filename" and "contentType" property in current model and its base models
+      let type: SdkModelType | undefined = property.type;
+      while (type !== undefined) {
```

Comment:
I tend to let TCGC return the file related info, e.g. filename, headers(content-type or other headers), and we just use it. I'm fine if we handle it by ourselves currently, we can discuss later.

Reply:
1. it could be in base class like `File`
2. it could be in multiple base class, if there is multiple `FileA extends FileB`, `FileB extends FileC`

Yeah, we may (or may not) be able to use the SdkModelProperty directly. E.g. I don't know what the SDK should generate, if typespec write
```
model FileA extends File {
  filename: "name1" | "name2"
}
```

Good news is that lint appears to error if adding any other property (other than this 3).